python/lsst/validate/drp/calcSrd.py

```diff
@@ -259,18 +267,225 @@ def computeWidths(array):
+        rmsDistMas, annulus, magrange
+
+    Notes
+    -----
+    This table below is provided in this package in the `srdSpec.py` file.
+
+    LPM-17 dated 2011-07-06
+
+    *The relative astrometry*
+    Specification: The rms of the astrometric distance distribution for
+        stellar pairs with separation of D arcmin (repeatability)
+        will not exceed AMx milliarcsec (median distribution for a large number
+        of sources). No more than AFx % of the sample will deviate by more than
+        ADx milliarcsec from the median. AMx, AFx, and ADx are specified for
+        D=5, 20 and 200 arcmin for x= 1, 2, and 3, in the same order (Table 18).
+
+    The three selected characteristic distances reflect the size of an
+    individual sensor, a raft, and the camera. The required median astrometric
+    precision is driven by the desire to achieve a proper motion accuracy of
+    0.2 mas/yr and parallax accuracy of 1.0 mas over the course of the survey.
+    These two requirements correspond to relative astrometric precision for a
+    single image of 10 mas (per coordinate).
+
+    ================== =========== ============ ============
+         Quantity      Design Spec Minimum Spec Stretch Goal
+    ------------------ ----------- ------------ ------------
+    AM1 (milliarcsec)        10          20            5
+    AF1 (%)                  10          20            5
+    AD1 (milliarcsec)        20          40           10
+
+    AM2 (milliarcsec)        10          20            5
+    AF2 (%)                  10          20            5
+    AD2 (milliarcsec)        20          40           10
+
+    AM3 (milliarcsec)        15          30           10
+    AF3 (%)                  10          20            5
+    AD3 (milliarcsec)        30          50           20
+    =================  =========== ============ ============
+    Table 18: The specifications for astrometric precision.
+    The three blocks of values correspond to D=5, 20 and 200 arcmin,
+    and to astrometric measurements performed in the r and i bands.
+    """
+
+    # Default is specified here separately because defaults that are mutable
+    # get overridden by previous calls of the function.
+    if magrange is None:
+        magrange = [17.0, 21.5]
+
+    # First we make a list of the keys that we want the fields for
+    importantKeys = [groupView.schema.find(name).key for
+                     name in ['id', 'coord_ra', 'coord_dec',
+                              'object', 'visit', 'base_PsfFlux_mag']]
+
+    # Includes magrange through closure
+    def magInRange(cat):
+        mag = cat.get('base_PsfFlux_mag')
+        w, = np.where(np.isfinite(mag))
+        medianMag = np.median(mag[w])
+        return magrange[0] <= medianMag and medianMag < magrange[1]
+
+    groupViewInMagrange = groupView.where(magInRange)
+
+    # List of lists of id, importantValue
+    matchKeyOutput = [x.get(y) for y in importantKeys for x in groupViewInMagrange.groups]
```

Comment:
I'd prefer slightly more descriptive temporary variables than x and y, especially in a context like this where those particular variables imply coordinates.  Even "k" and "g".